tutorials/haskell/generics/tutorial.md

```diff
@@ -0,0 +1,584 @@
+---
+title: GHC Generics Explained
+published: 2017-02-01
+ghc: 8.0.1
+lts: 7.16
+language: haskell
+author: Mark Karpov
+author-name: Mark Karpov
+description: This tutorial will get you up to speed with GHC generics quickly.
+---
+
+This tutorial will get you up to speed with GHC generics quickly. It should
+be noted that generics is not something academic and useless, quite the
+contrary, it's a very pragmatic way to reduce the amount of boilerplate (and
+associated with it errors) in your functional code with minimal mental
+overhead. In fact, by the time you get to the end of the tutorial, I hope
+you'll agree that generics are easy to use and easy “to get” as well.
+
+Let's start from the beginning. What is generics? **Generics**, like in
```

Comment:
I'll agree on this, although I talked about generics as a language feature, but perhaps if the language feature's name is in plural form, this version is more correct.